### PR TITLE
[Enhancement] Enable read pindex page by page

### PR DESCRIPTION
**Pull request by sevev**
## Why I'm doing:
Reading pindex by shard may lead to I/O amplification, while reading by page can reduce I/O amplification.

## What I'm doing:
 Enable read pindex page by page

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by wyb — DISMISSED** [review]

**Review by luohaha — DISMISSED** [review]

**Review by wyb — DISMISSED** [review]

**Review by luohaha — DISMISSED** [review]

File: be/test/storage/persistent_index_test.cpp [hunk 1]
```diff
@@ -44,7 +44,10 @@ struct PersistentIndexTestParam {
 class PersistentIndexTest : public testing::TestWithParam<PersistentIndexTestParam> {
 public:
     virtual ~PersistentIndexTest() {}
-    void SetUp() override { config::enable_pindex_compression = GetParam().enable_pindex_compression; }
+    void SetUp() override {
+        config::enable_pindex_compression = GetParam().enable_pindex_compression;
+        config::enable_pindex_read_by_page = true;
```
Comment (kevincai) [inline be/test/storage/persistent_index_test.cpp:49]: should restore the configvar value in TearDown(), avoid pollute other cases which the behavior might be affected by this configvar.

Comment (sevev) [reply be/test/storage/persistent_index_test.cpp:49]: > should restore the configvar value in TearDown(), avoid pollute other cases which the behavior might be affected by this configvar.

This behavior will be set as default behavior in future production env and this pr refrains from altering the default behavior to mitigate potential impacts from minor version upgrades. In actuality, I intend to conduct unit testing with this new behavior, which is why I have opted not to reset the configuration in TearDown().

Comment (kevincai) [reply be/test/storage/persistent_index_test.cpp:49]: a good practice is to restore the env to exactly the same as the ut enters. if in future the default value is going to be `true`, set it in config.h.

if you want to try it first in ut, set it in be_test.conf or in ut main().

No one knows your intention behind this without any comments.

B.T.W. our CI is running in parallel, each UT suite is running in a separate process, so your intention in this mode doesn't work. Make the code explicitly reflect your design intention.

**Review by kevincai — APPROVED** [review]

**Review by luohaha — APPROVED** [review]

File: be/test/storage/persistent_index_test.cpp (hunk 1 above)
Comment (sevev) [reply be/test/storage/persistent_index_test.cpp]: > a good practice is to restore the env to exactly the same as the ut enters. if in future the default value is going to be `true`, set it in config.h.
> 
> if you want to try it first in ut, set it in be_test.conf or in ut main().
> 
> No one knows your intention behind this without any comments.
> 
> B.T.W. our CI is running in parallel, each UT suite is running in a separate process, so your intention in this mode doesn't work. Make the code explicitly reflect your design intention.

A good advice, changed in https://github.com/StarRocks/starrocks/pull/42870/commits/bfa05915a117550d36f0b13b313ec08bb3ea01fb, PTAL again, thx.

**Review by wyb — APPROVED** [review]

**Pull request merged**